### PR TITLE
Exclude affinity-setting logic from minimal build

### PR DESCRIPTION
**Pull request by RandySheriffH**
Comment out the affinity-setting logic which introduced an unnecessary binary size increase for the minimal build.

### CODE REVIEW

**Comment by skottmckay** [comment]
>   ORT_TRY {

Is ReadThreadAffinityConfig intended to be file scope? Not sure linker can throw it away as is so put in an anonymous namespace or make static.

---
Refers to: onnxruntime/core/util/thread_utils.cc:23 in 5f03d4d. [](commit_id = 5f03d4d53cff09d6071a53e59bf3ca8ac0d21346, deletion_comment = False)

File: onnxruntime/core/util/thread_utils.cc [hunk 1]
```diff
@@ -87,6 +87,10 @@ CreateThreadPoolHelper(Env* env, OrtThreadPoolParams options) {
   }
   // override affinity setting if specified from customer
   if (!options.affinity_str.empty()) {
+#if defined(ORT_MINIMAL_BUILD) || defined(ORT_EXTENDED_MINIMAL_BUILD)
+    ORT_THROW("Setting thread affinity is not implemented in this build.");
+    return nullptr;
+#else
     to.affinities = ReadThreadAffinityConfig(options.affinity_str);
```
Comment (skottmckay) [inline onnxruntime/core/util/thread_utils.cc:94]: Is the implementation of ReadThreadAffinityConfig intended to be file scope? Not sure linker can throw it away as-is, so put in an anonymous namespace or make static if it's meant to be file scope.

Comment (RandySheriffH) [reply onnxruntime/core/util/thread_utils.cc:94]: [now](https://dev.azure.com/onnxruntime/onnxruntime/_build/results?buildId=835135&view=logs&j=cbd9a98b-c201-5ad2-0e23-4c99a370d00f&t=adf49265-58d4-533c-580f-c52bffeeb92d)
[before](https://dev.azure.com/onnxruntime/onnxruntime/_build/results?buildId=832996&view=logs&j=cbd9a98b-c201-5ad2-0e23-4c99a370d00f&t=adf49265-58d4-533c-580f-c52bffeeb92d)

Comment (skottmckay) [reply onnxruntime/core/util/thread_utils.cc:94]: Cool. Would be good to fix though to make it clear the function is intended to have file scope.

Comment (RandySheriffH) [reply onnxruntime/core/util/thread_utils.cc:94]: Sure, done.

**Review by skottmckay — DISMISSED** [review]
:shipit:

**Review by skottmckay — APPROVED** [review]

**Pull request merged**